### PR TITLE
Fix Jest tests not runnable

### PR DESCRIPTION
**Pull request by soyelmnd**
This is an attempt to fix the annoying issue with Jest tests not runnable in Intellij if we import the project via bazel (importing just `web/` should be fine I reckon), and to save ppl from having to use some `fkn_` scripts. More info in [this Slack thread](https://canva.slack.com/archives/C7883Q2JH/p1656644668287419?thread_ts=1581580515.025300&cid=C7883Q2JH) and https://github.com/bazelbuild/intellij/issues/3629.

I've tried on IJ 2022.1.4 using the bazel plugin from s3 (the _before_) vs. the updated one from this branch [0] (the _after_) 👇 

The `runConfigurations.xml` file seems to be ~~regenerated~~ updated every time Intellij completes its start-up [1]. With the updated plugin, `JestRunConfigurationProvider` isn't included in the `ignoredProducts` and the Jest tests run just fine 🎉 Please note the updated plugin, however, doesn't remove the line, it just doesn't add it in, hence a manual removal is necessary, e.g.

```bash
sed -i '' '/JestRunConfigurationProducer/d' .ijwb/.idea/runConfigurations.xml
```

| before | after |
|-|-|
| before | after |
| b | a |

- [0] The built plugin [ijwb_bazel.zip](https://canva.slack.com/archives/C7883Q2JH/p1656646150006419?thread_ts=1581580515.025300&cid=C7883Q2JH) FYI, was built using [Uri's suggestion](https://canva.slack.com/archives/C7883Q2JH/p1656646150006419?thread_ts=1581580515.025300&cid=C7883Q2JH)
- [1] Sorry I haven't dug into the plugin code to figure out exactly when the `runConfigurations.xml` is ~~generated~~ updated (nor do I understand what those producers are) 😁 but as I've tried it with the existing plugin, 10 times out of 10, the `JestRunConfigurationProvider` line comes back after IJ completes its initial start-up (before indexing it seems?)

### CODE REVIEW

**Comment by uri-canva** [comment]
Note this repo is a public fork so people reading the PRs might not be able to access the links.

File: javascript/src/com/google/idea/blaze/javascript/run/producers/NonBlazeProducerSuppressor.java
Comment (uri-canva) [inline javascript/src/com/google/idea/blaze/javascript/run/producers/NonBlazeProducerSuppressor.java]: I think you forgot to remove it here:
```suggestion
              // Not suppressing JestRunConfigurationProduce since that prevents Jest tests from
              // running properly, see https://github.com/bazelbuild/intellij/issues/3629
              // JestRunConfigurationProducer.class,
```

**Review by uri-canva — APPROVED** [review]

**Comment by uri-canva** [comment]
Since the file is regenerated does that mean that we don't need to do anything to fix the installations of people who have run the plugin without this fix?

**Comment by uri-canva** [comment]
Also note that we don't merge our PRs into `master`, we merge them into a release branch, so you'll have to create a new one, then update the target branch of this PR. Cherry pick the commits from the previous release branch before you merge this PR.

File: javascript/src/com/google/idea/blaze/javascript/run/producers/NonBlazeProducerSuppressor.java
Comment (soyelmnd) [reply javascript/src/com/google/idea/blaze/javascript/run/producers/NonBlazeProducerSuppressor.java]: Oops, well spotted, thanks Uri 👍 you can tell I lazily made this commit using github editor, apologise 😂

Confirm no other changes
image

**Review by soyelmnd — COMMENTED** [review]
> Since the file is regenerated does that mean that we don't need to do anything to fix the installations of people who have run the plugin without this fix?

That's a _great_ question Uri, exactly what I didn't think about, thanks. As I've just checked, seemed like the line did not get removed when I swapped the new plugin in (it just didn't add the line back in after a manual removal), which means the assumption that the whole file gets regenerated sounds incorrect. I'll update the PR desc with this note.

And to answer the question: yes, we do need to manually fix up the `runConfiguration.xml` after installing the plugin, e.g. 👇  

```
sed -i '' '/JestRunConfigurationProducer/d' .ijwb/.idea/runConfigurations.xml
```

> we merge them into a release branch

Ah yup, thanks, will do 👍 

--

I'll keep using and monitoring the plugin for a while (thinking this week) before merging this, will keep you posted. Thanks Uri

**Comment by soyelmnd** [comment]
Just an update that I've been testing this plugin for a couple of days and seems to be ok, haven't noticed anything strange. I'm looking to merge this later today and will get the plugin published later this week or early next week.

EDIT: I'll leave this PR with you Serge :) thanks

**Review by serge-belov — APPROVED** [review]

**Comment by serge-belov** [comment]
hey @soyelmnd, to recap our discussion in Slack here: if that's OK with you, I'd like to wait with the merge till the update to 2022.1.4 and hopefully the fix for the 3rd party source loading issue that I'm trying to understand and fix. The reason is mainly me switching from the branch-per-release to a trank-based approach for this repo so at the same time. If you don't mind, I'll ping you sometime next week once ready to merge from my end.

**Pull request merged**
